### PR TITLE
Integrate to do

### PR DESCRIPTION
**Pull request by allwelly**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change that updates code, but does not affect functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

pressed + buttons, to type and add todo items
empty items can be added to the list, might wanna prevent this in the future
pressed edit button to switch to editing mode and edit the text, changes will be reflected after tapping the check button
ticked items will become un-ticked after being edited

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: front-end/virtual_pet/lib/widgets/todo_item.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Todo item class
+class Todo {
+  String name;
+  bool checked;
+
+  Todo({
+    required this.name, 
+    required this.checked});
```
Comment (mebesius) [inline front-end/virtual_pet/lib/widgets/todo_item.dart:8]: Can we move the `});` to the next line please?

File: front-end/virtual_pet/lib/widgets/todo_tile.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import 'package:flutter/material.dart';
+import 'package:virtual_pet/widgets/todo_item.dart';
+
+// Todo item tile widget
+class TodoItemTile extends StatefulWidget {
+  final Todo todo;
+  final Function (Todo todo) onChecked;
```
Comment (mebesius) [inline front-end/virtual_pet/lib/widgets/todo_tile.dart:7]: Can we remove the ` ` between the end of the `Function` type and `(` so it is consistent with the one below and with function definitions/calls more generally?

**Review by mebesius — APPROVED** [review]
Good job @allwelly ! It looks really good and is working well from my testing :) I just have a few little comments but since the demo is tomorrow, I am still going to approve the PR since they are so minor and the code is functionally correct

**Review by lourw — APPROVED** [review]
Approving for brevity

**Pull request merged**